Source/WebKit/UIProcess/WebProcessProxy.cpp

```diff
@@ -1561,8 +1561,17 @@ void WebProcessProxy::sendProcessDidResume(ResumeReason)
         send(Messages::WebProcess::ProcessDidResume(), 0);
 }
 
+void WebProcessProxy::setThrottleStateForTesting(ProcessThrottleState state)
+{
+    m_areThrottleStateChangesEnabled = true;
+    didChangeThrottleState(state);
+    m_areThrottleStateChangesEnabled = false;
```

Comment:
This wasn't needed before https://github.com/WebKit/WebKit/pull/5759 because `ProcessThrottler` would rarely fire `didChangeThrottleState()` on macOS. Now that it fires frequently, it overwrites the state after tests call `setThrottleStateForTesting()`.